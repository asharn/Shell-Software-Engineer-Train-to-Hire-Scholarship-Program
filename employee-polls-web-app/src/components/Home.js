import React from 'react';
import { connect } from "react-redux";
import { useLocation, useNavigate, useParams } from "react-router-dom";

  
const withRouter = (Component) => {
  const ComponentWithRouterProp = (props) => {
    let location = useLocation();
    let navigate = useNavigate();
    let params = useParams();
    return <Component {...props} router={{ location, navigate, params }} />;
  };

  return ComponentWithRouterProp;
};

const Home = () => {
  return (
    <div
      style={{
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        height: '80vh'
      }}
    >
      <h1>Welcome to Employee Polls</h1>
    </div>
  );
};
  
const mapStateToProps = ({ authedUser }) => ({
  userLoggedIn: authedUser !== null,
});
export default withRouter(connect(mapStateToProps)(Home));