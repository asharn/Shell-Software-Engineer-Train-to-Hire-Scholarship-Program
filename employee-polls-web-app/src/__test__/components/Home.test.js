import { render, screen } from '@testing-library/react';
import Home from '../../components/Home';
import configureStore from 'redux-mock-store';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

const mockStore = configureStore([]);

describe('Snapshot testing for Home component.', () => {
    let store;
    let component;

    beforeEach(() => {
      store = mockStore({
        myState: 'sample text',
      });

      component = renderer.create(
        <Provider store={store}>
          <Home />
        </Provider>
      );
    });
    
  test('renders Home component with toMatchSnapshot.', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renders App component screen debug.', () => {
    render(<Provider store={store}><Home /></Provider>);
    screen.debug();
   });


})