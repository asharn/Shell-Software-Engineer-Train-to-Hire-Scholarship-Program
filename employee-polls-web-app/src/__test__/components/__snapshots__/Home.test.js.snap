// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing for Home component. renders Home component with toMatchSnapshot. 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "direction": "column",
      "display": "grid",
      "height": "75vh",
      "justifyContent": "center",
    }
  }
>
  <h1>
    Welcome to Home page of Employee Polls. 
  </h1>
  <h2>
    This is public page, so no need to sign in. Anyone can visit.
  </h2>
</div>
`;
